cli/src/pcluster/validators/cluster_validators.py

```diff
@@ -278,6 +281,18 @@ def validate_networking_requirements(
                 FailureLevel.WARNING,
             )
 
+    def validate_allocation_strategy(self, compute_resource_name: str, capacity_type: Enum, allocation_strategy: Enum):
+        """On-demand Capacity type only supports "lowest-price" allocation strategy."""
+        if (
+            capacity_type == cluster_config.CapacityType.ONDEMAND
+            and allocation_strategy == cluster_config.AllocationStrategy.CAPACITY_OPTIMIZED
+        ):
+            self._add_failure(
+                f"Compute Resource {compute_resource_name} is using an OnDemand CapacityType. OnDemand CapacityType "
+                "can only use 'lowest-price' allocation strategy.",
```

Comment:
use the enum instead of the 'lowest-price' string, so that if value changes there is only one place where it has to be changed and message will be still valid